File: features/lead/src/main/java/com/nivasafinance/features/leadlender/service/LeadLenderWriteService.java
package com.nivasafinance.features.leadlender.service;

import com.nivasafinance.features.leadlender.dto.CreateLeadLenderRequest;
import com.nivasafinance.features.leadlender.dto.CreateLeadLenderResponse;
import com.nivasafinance.features.leadlender.dto.RejectLeadLenderRequest;
import com.nivasafinance.features.leadlender.dto.UpdateLeadLenderRequest;

import java.util.UUID;

public interface LeadLenderWriteService {
    
    CreateLeadLenderResponse createLeadLender(UUID leadIdentifier, CreateLeadLenderRequest request);
    
    void updateLeadLender(UUID leadIdentifier, UUID lenderIdentifier, UpdateLeadLenderRequest request);
    
    void rejectLeadLender(UUID leadIdentifier, UUID lenderIdentifier, RejectLeadLenderRequest request);
}

File: features/lead/src/main/java/com/nivasafinance/features/leadlender/service/LeadLenderWriteService.kt
package com.nivasafinance.features.leadlender.service

import com.nivasafinance.features.leadlender.dto.CreateLeadLenderRequest
import com.nivasafinance.features.leadlender.dto.CreateLeadLenderResponse
import com.nivasafinance.features.leadlender.dto.RejectLeadLenderRequest
import com.nivasafinance.features.leadlender.dto.UpdateLeadLenderRequest
import java.util.UUID

interface LeadLenderWriteService {
    fun createLeadLender(leadId: UUID, request: CreateLeadLenderRequest): CreateLeadLenderResponse
    fun updateLeadLender(lenderId: UUID, request: UpdateLeadLenderRequest)
    fun rejectLeadLender(lenderId: UUID, request: RejectLeadLenderRequest)
}
